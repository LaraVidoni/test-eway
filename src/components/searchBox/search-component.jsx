import React, { Component } from "react";
import messages from "./messages";
import "./styles.css";

let timeout;
const PredictiveElement = ({ children, ...rest }) => <li className="predictive-element" {...rest}>{children}</li>

export default class SearchComponent extends Component {
  state = {
    searchValue: ``,
    words: [], 
    hidePredictiveSearch: true
  }

  render() {
    let language = this.props.language || 'en';
    let resources = messages[language];
    const { words, hidePredictiveSearch, searchValue } = this.state;
    const classNames = hidePredictiveSearch ? 'hide' : 'show';
    return (
      <div className="search-component">
        <form className="search-form" onSubmit={this.handleSubmit}>
          <div className="search-container">
            <div className="search-bar-container">
              <input
                type="text"
                className="search-bar"
                placeholder={resources.placeHolder}
                onChange={this.filterWords}
                onFocus={this.showSearchContainer}
                value={searchValue}
                onBlur={this.hideSearchContainer}
              />
              <ul className={`predictive-list ${classNames}`}>
                {words.map((word, index) => <PredictiveElement onMouseDown={this.handleSelectPreditiveElement} key={index}>{word}</PredictiveElement>)}
              </ul>
            </div>
            <input
              type="submit"
              className="search-button"
              value={resources.searchButton}
            />
          </div>
        </form>
      </div>
    )
  }

  handleSubmit = (value) => {
    window.location.replace(`/Eway/${this.props.language}/Shopping/Search/SearchByKeyword/${typeof value === "string" ? value : value.currentTarget[0].value}`);
  }

  handleSelectPreditiveElement = (e) => {
    const value = e.currentTarget.textContent;
    this.getWords(value);
    this.setState({
      searchValue: value
    });
    this.handleSubmit(value);
  }

  hideSearchContainer = () => {
    this.setState({
      hidePredictiveSearch: true
    })
  }

  showSearchContainer = () => {
    this.setState({
      hidePredictiveSearch: false
    })
  }

  filterWords = (e) => {
    const value = e.currentTarget.value.toLowerCase()
    // Updates the words. 
    this.getWords(value);
    // Updates the search input
    this.setState({
      searchValue:value,
    })
  }

  getWords = function (searchValue) {
    clearTimeout(timeout);
    timeout = setTimeout(() => {
      if (!searchValue) {
        return;
      }
      const terms = this.getTermsList()
      //const words = terms.filter(term => term.includes(searchValue))
      this.setState({
        words: terms,
        hidePredictiveSearch: false
      })
    }, 500);
  }

  getTermsList = () => {
    return [
        "hello",
        "bonjour",
        "hey",
        "hola",
        "hi",
        "maybe",
        "this",
        "is",
        "text",
        "zebra",
        "internationalization"
      ]
    
    //   fetch('https://api.thecatapi.com/v1/images/search?format=json', {mode: 'no-cors', headers: new Headers({'Content-Type': 'application/json'}, {'x-api-key': '17d94b92-754f-46eb-99a0-65be65b5d18f'})})
    //     .then(function (response) {
    //       console.log(response);
    //       return response.json();
    //     })
    //     .then(function (myJson) {
    //       console.log(JSON.stringify(myJson));
    //     });
  }
}