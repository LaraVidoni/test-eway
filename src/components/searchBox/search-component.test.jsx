import React from 'react';
import {SearchComponent} from './search-component';
import renderer from "react-test-renderer";

it('renders in english (as default language)', () => {
  const div = document.createElement('div');
  const tree = renderer.create(<SearchComponent />, div).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders in english', () => {
  const div = document.createElement('div');
  const tree = renderer.create(<SearchComponent language="en" />, div).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders in french', () => {
  const div = document.createElement('div');
  const tree = renderer.create(<SearchComponent language="fr" />, div).toJSON();
  expect(tree).toMatchSnapshot();
});