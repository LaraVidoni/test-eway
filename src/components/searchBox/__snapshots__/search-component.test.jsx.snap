// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders in english (as default language) 1`] = `
<div
  className="search-component"
>
  <form
    action="/Eway/en/Shopping/Search/SearchByKeyword"
    method="post"
  >
    <div
      className="search-container"
    >
      <input
        className="search-bar"
        placeholder="Enter Keyword or Product Number here"
        type="text"
      />
      <input
        className="search-button"
        type="submit"
        value="Search"
      />
    </div>
  </form>
</div>
`;

exports[`renders in english 1`] = `
<div
  className="search-component"
>
  <form
    action="/Eway/en/Shopping/Search/SearchByKeyword"
    method="post"
  >
    <div
      className="search-container"
    >
      <input
        className="search-bar"
        placeholder="Enter Keyword or Product Number here"
        type="text"
      />
      <input
        className="search-button"
        type="submit"
        value="Search"
      />
    </div>
  </form>
</div>
`;

exports[`renders in french 1`] = `
<div
  className="search-component"
>
  <form
    action="/Eway/fr/Shopping/Search/SearchByKeyword"
    method="post"
  >
    <div
      className="search-container"
    >
      <input
        className="search-bar"
        placeholder="Entrez un mot-clÃ© ou un # produit"
        type="text"
      />
      <input
        className="search-button"
        type="submit"
        value="Rechercher"
      />
    </div>
  </form>
</div>
`;
